src/ogr/feature.jl

```diff
@@ -533,11 +572,21 @@ function setfield!(feature::Feature, i::Integer, value::Int32)::Feature
     return feature
 end
 
+function setfield!(feature::Feature, i::Integer, value::UInt16)
+    GDAL.ogr_f_setfieldinteger(feature.ptr, i, Int32(value))
```

Comment:
Out of curiosity: would it still work if we didn't convert `value` to `Int32` (i.e. would the ccall in `GDAL.ogr_f_setfieldinteger(...)` do it automatically)? Asking also because we don't seem to do it for `value::Int16` on L580.

The tests seem to indicate that we need to do appropriate widening / "type promotions" -- are we doing it here because the ccall conversions aren't the ones we expect? Similar questions for the conversions below.

Reply:
Good point, it seems to work fine without them indeed, I've now unified all the types for the `setfield!` operation. 

In the tests I wanted to make sure that in the Julia ecosystem the subtype is used, even though the GDAL API only sees it as metadata. GDAL does check whether the value fits in the subtype in `ogr_f_setfieldx`.

Reply:
Thank you!